lib/delayed/worker.rb

```diff
@@ -114,38 +109,41 @@ def work_off(num = 100)
         break if stop? # leave if we're exiting
       end
 
-      [success, failure].map(&:value)
+      [success.value, total - success.value]
     end
 
     def run_thread_callbacks(job, &block)
       self.class.lifecycle.run_callbacks(:thread, self, job, &block)
     end
 
     def run(job)
-      metadata = {
-        status: 'RUNNING',
-        name: job.name,
-        run_at: job.run_at,
-        created_at: job.created_at,
-        priority: job.priority,
-        queue: job.queue,
-        attempts: job.attempts,
-        enqueued_for: (Time.current - job.created_at).round,
-      }
-      job_say job, metadata.to_json
-      run_time = Benchmark.realtime do
-        Timeout.timeout(max_run_time(job).to_i, WorkerTimeout) do
-          job.invoke_job
+      run_thread_callbacks(job) do
+        metadata = {
+          status: 'RUNNING',
+          name: job.name,
+          run_at: job.run_at,
+          created_at: job.created_at,
+          priority: job.priority,
+          queue: job.queue,
+          attempts: job.attempts,
+          enqueued_for: (Time.current - job.created_at).round,
+        }
+        job_say job, metadata.to_json
+        run_time = Benchmark.realtime do
+          Timeout.timeout(max_run_time(job).to_i, WorkerTimeout) do
+            job.invoke_job
+          end
+          job.destroy
         end
-        job.destroy
+        job_say job, format('COMPLETED after %.4f seconds', run_time)
       end
-      job_say job, format('COMPLETED after %.4f seconds', run_time)
       true # did work
     rescue DeserializationError => e
       job_say job, "FAILED permanently with #{e.class.name}: #{e.message}", 'error'
 
       job.error = e
       failed(job)
+      false # work failed
```

Comment:
This was a bug that improperly reported deserialization errors in the `success` number, which only impacted the logging output.